mitodl/mit-open
# Upgrade tika and disable OCR via headers

### What are the relevant tickets?
Closes #393 

### Description (What does it do?)
- Disables OCR via the request header (OCR is enabled on RC/production but not locally, and drastically increases processing time)
- Makes the OCR header setting configurable as a setting (in case we ever want to change it)
- Upgrades tika to the same version used in rc/prod
- Upgrades the tika python library


### How can this be tested?
Try the following under two sets of settings:

Set 1: 
```
TIKA_SERVER_ENDPOINT=http://tika:9998/
TIKA_ACCESS_TOKEN=
```

Set 2:
```
TIKA_SERVER_ENDPOINT=<same as on heroku rc>
TIKA_ACCESS_TOKEN=<same as on heroku rc>
```

Run the following:

```python
from learning_resources.etl.utils import extract_text_from_url
extract_text_from_url("https://ocw.mit.edu/courses/18-725-algebraic-geometry-fall-2015/ec341c7a2524e5dba7c3e939f322613a_MIT18_725F15_notes.pdf")
```

In both cases you should get a response back within a few seconds or less.
